angular-http-server.js

```diff
@@ -1,14 +1,45 @@
 #!/usr/bin/env node
 
-var http = require("http");
 var fs = require("fs");
 var argv = require('minimist')(process.argv.slice(2));
 
-var server = http.createServer(function (req, res) {
+var server;
+if (argv.ssl || argv.https) {
+    var pem = require('pem');
+    var https = require('https');
+    pem.createCertificate({days: 1, selfSigned: true}, function(err, keys) {
+        var options = {
+            key: keys.serviceKey,
+            cert: keys.certificate,
+            rejectUnauthorized: false
+        };
+        server = https.createServer(options, requestListener);
+        start();
+    });
+}
+else {
+    var http = require("http");
+    server = http.createServer(requestListener);
+    start();
+}
+
+function requestListener(req, res) {
     // console.log(req.url);
     var url = req.url.split('?')[0]
     var possibleFilename = url.slice(1) || "dummy";
 
+    if (argv.cors) {
+      res.setHeader('Access-Control-Allow-Origin', '*');
+      res.setHeader('Access-Control-Request-Method', '*');
+      res.setHeader('Access-Control-Allow-Methods', 'OPTIONS, GET');
+      res.setHeader('Access-Control-Allow-Headers', 'authorization, content-type');
+      if ( req.method === 'OPTIONS' ) {
+        res.writeHead(200);
+        res.end();
```

Comment:
My bad, the CORS stuff wasn't supposed to be a part of this PR -- although I guess it's not necessarily a bad addition.

Anyway, the OPTIONS method identifies the allowed methods, which are contained in the Allow headers. Responses to OPTIONS methods also have a Content-Length of 0. When using CORS, a preflighted request with the OPTIONS method is sent so that the server can check for CORS headers.

To enable CORS server-wide, the server needs to respond with the Allow headers and a 200 code. This is all that is necessary on an OPTIONS request, which is why the response ends immediately with zero content.